firmware/ventilator-controller-stm32/Core/Inc/Pufferfish/Protocols/Messages.tpp

```diff
@@ -98,15 +98,17 @@ MessageStatus MessageReceiver<Message, num_descriptors>::transform(
 
 // MessageSender
 
-template <typename Message, size_t num_descriptors>
-MessageSender<Message, num_descriptors>::MessageSender(
-    const Util::ProtobufDescriptors<num_descriptors> &descriptors)
+template <typename Message, typename TaggedUnion, size_t num_descriptors>
+MessageSender<Message, TaggedUnion, num_descriptors>::MessageSender(
+    Util::ProtobufDescriptors<num_descriptors> &descriptors)
     : descriptors_(descriptors) {}
 
-template <typename Message, size_t num_descriptors>
+template <typename Message, typename TaggedUnion, size_t num_descriptors>
 template <size_t output_size>
-MessageStatus MessageSender<Message, num_descriptors>::transform(
-    const Message &input_message, Util::ByteVector<output_size> &output_buffer) const {
+MessageStatus MessageSender<Message, TaggedUnion, num_descriptors>::transform(
+    const TaggedUnion &input_payload, Util::ByteVector<output_size> &output_buffer) const {
+  Message input_message;
+  input_message.payload = input_payload;  // NOLINT(cppcoreguidelines-pro-type-union-access)
```

Comment:
As discussed in meeting, we'll add a copy assignment operator to `TaggedUnion` where assigning the `value` member is suppressed with a `NOLINTNEXTLINE` (and we'll need to add a unit test or two of a TaggedUnion where the union type has constituent types of different sizes (e.g. a union of a `uint8_t` and a `uint16_t`)